pkg/testing/linearizer/linearizer.go

```diff
@@ -0,0 +1,252 @@
+package linearizer
+
+import (
+	"fmt"
+	"math/rand"
+	"sort"
+	"sync"
+	"sync/atomic"
+	"testing"
+)
+
+/*
+   A controller has this sequence: Get, Create, and Update
```

Comment:
`Create` here assumes "Delete and Create" on type change or its purely "Create" when the item doesn't exist?

Reply:
this is just an example to demonstrate how the framework generates the API call sequence, we could replace the Create/Delete with DoFoo/DoBar here